src/dsp_tools/models/connection.py

```diff
@@ -27,240 +29,260 @@ def check_for_api_error(response: requests.Response) -> None:
-
-    @property
-    def token(self) -> str:
-        """Get the token of this connection"""
-        return self._token
+        self.token = result["token"]
 
     def start_logging(self) -> None:
         """Start writing every API call to a file"""
-        self._log = True
+        self.log = True
+        self.log_directory.mkdir(exist_ok=True)
 
-    def stop_logging(self):
+    def stop_logging(self) -> None:
         """Stop writing every API call to a file"""
-        self._log = False
+        self.log = False
 
     def logout(self) -> None:
-        """
-        Performs a logout
-        :return: None
-        """
-
-        if self._token is not None:
+        """Delete the session token"""
+        if self.token:
             response = requests.delete(
-                self._server + "/v2/authentication",
-                headers={"Authorization": "Bearer " + self._token},
+                self.server + "/v2/authentication",
+                headers={"Authorization": "Bearer " + self.token},
                 timeout=20,
             )
             check_for_api_error(response)
-            self._token = None
-
-    def __del__(self):
-        pass
-        # self.logout()
-
-    def post(self, path: str, jsondata: Optional[str] = None):
+            self.token = None
+
+    def write_log_file(
+        self,
+        method: str,
+        route: str,
+        headers: dict[str, str],
+        jsondata: Optional[str],
+        params: Optional[dict[str, Any]],
+        response: requests.Response,
+    ) -> None:
         """
-        Post Json data to a given server using a HTTP POST request
-        :param path: Path of RESTful route
-        :param jsondata: Valid JSON as string
-        :return: Response from server
+        If logging is enabled (self.log == True), write the request and response to a file.
+
+        Args:
+            method: HTTP method (POST, GET, PUT, DELETE)
+            route: route of DSP-API that was called
+            headers: headers of the HTTP request
+            jsondata: data sent to the server
+            response: response of the server
         """
+        if not self.log:
+            return
+        if response.status_code == 200:  # pylint: disable=duplicate-code
+            _return = response.json()
+        else:
+            _return = {"status": response.status_code, "message": response.text}
+        logobj = {
+            "DSP server": self.server,
+            "route": route,
+            "method": method,
+            "headers": headers,
+            "params": params,
+            "body": json.loads(jsondata) if jsondata else None,
+            "return-headers": dict(response.headers),
+            "return": _return,
+        }
+        filename = f"{datetime.now().strftime('%Y-%m-%d %H.%M.%S.%f')} {method} {route.replace('/', '_')}.json"
+        with open(self.log_directory / filename, "w", encoding="utf-8") as f:
+            json.dump(logobj, f, indent=4)
```

Comment:
I was wondering two things here:

1. Should this really be one file per request (potentially very many files), or would it be better to have a single file per run, containing all requests (potentially a huge file; plus you'd need  to be able to append efficiently). I'm not sure to be honest.
2. is JSON the best format for this? Generally, JSON is nice most times - but the most relevant parts of this, request and response, are both JSON themselves; so you end up with JSON in JSON which greatly reduces readability.  
My first intuition was to suggest simply using a .txt file, but that's not nice either.  
Then I thought, you could try to emulate a .http file (seems to be adopted quite widely by now, see [here](https://marketplace.visualstudio.com/items?itemName=humao.rest-client), [here](https://www.jetbrains.com/help/idea/http-client-in-product-code-editor.html), [here](https://learn.microsoft.com/en-us/aspnet/core/test/http-files?view=aspnetcore-8.0), etc.). The nice thing would be that you get syntax highlighting, if properly done, the requests can even be executed, and the next request can simply be appended after a divider. The downside is that it would be quite a bit more complex to implement.

Reply:
I tried it out with the `create` and `xmlupload` commands, and I found it convenient with 1 file/request, and also the JSON format was well readable. If we are going to use this feature a lot, then I can think about .http files, which is quite cool, but not worth the effort right now.

Reply:
fair enough. But then maybe some mechanism needs to be in place for large uploads (say Vij's bulk uploads or something of that size) to not swamp the file system with ten-thousands of files? or do you think that would not be an issue?

Reply:
This is really an issue, but I'd like to put the burden on the user. Because I cannot think of an elegant way how to prevent too many files to be written. 
1. Detect if there are too many: in addition to sending requests and writing dump files, this class would have to measure how many MB are already in the dump directory. This measurement would take place before EVERY dump, which slows down the main program.
2. If the directory is full, what to do? Stop writing files? Deleting the old files?

I tend to be lazy here and appeal to the user's responsibility.

Reply:
As the user has to actively activate the dumping, I suppose the lazy approach is ok.

If you still wanted to go for it, here are some thoughts:
- I don't think disk storage space will be the biggest issue, instead what would be problematic much faster, is having too many small files in one directory.
- so the simple solution would be to create sub-directories
- a first step would be to have a subdirectory per run of DSP-Tools, so that the number of files does not accumulate
- and to go even further, you could even divide single runs up
- here, I don't think, creating datamodels would cause any issues, it's probably only the data upload.
- and for the data upload you parse the XML first, so you should know the number of resources to be created
- and you should be able to make a rough guess how many requests per resource there may be
- then you could say: if its less than x (I don't know... maybe 1000?) resources, no further subfolders are needed
- and if it's more, you could have subfolders `0`, `1`, etc. for every x resources
I don't think that would have big performance implications... but it's probably not worth the effort.